frontend/app/mocks/node.ts

```diff
@@ -1,21 +1,55 @@
+import jsonpatch, { JsonPatchError, type Operation } from 'fast-json-patch';
 import { HttpResponse, http } from 'msw';
 import { setupServer } from 'msw/node';
+import { z } from 'zod';
+
+import { db } from '~/mocks/db';
+
+const userIdSchema = z.string().uuid();
 
 const handlers = [
-  // prettier-ignore
-  http.get('https://api.example.com/users/*', () => {
+  http.get('https://api.example.com/users/:id', ({ params }) => {
+    const parsedUserId = userIdSchema.safeParse(params.id);
+    const userEntity = parsedUserId.success ? db.user.findFirst({ where: { id: { equals: parsedUserId.data } } }) : undefined;
+
+    if (!userEntity) {
+      return new HttpResponse('User Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+    }
+
     return HttpResponse.json({
-      firstName: 'John',
-      lastName: 'Maverick',
-      phoneNumber: '(555) 555-5555',
-      homeAddress: '123 Home Street',
-      mailingAddress: '123 Mailing Street',
-      preferredLanguage: 'fr'
-    })
+      id: userEntity.id,
+      firstName: userEntity.firstName,
+      homeAddress: userEntity.homeAddress,
+      lastName: userEntity.lastName,
+      mailingAddress: userEntity.mailingAddress,
+      phoneNumber: userEntity.phoneNumber,
+      preferredLanguage: userEntity.preferredLanguage,
+    });
   }),
 
-  http.patch('https://api.example.com/users/*', () => {
-    return new HttpResponse(null, { status: 204 });
+  http.patch('https://api.example.com/users/:id', async ({ params, request }) => {
+    const parsedUserId = userIdSchema.safeParse(params.id);
+    const userEntity = parsedUserId.success ? db.user.findFirst({ where: { id: { equals: parsedUserId.data } } }) : undefined;
+
+    if (!userEntity) {
+      return new HttpResponse('User Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+    }
+
+    try {
+      const { id, homeAddress, mailingAddress, phoneNumber, preferredLanguage } = userEntity;
+      const document = { homeAddress, mailingAddress, phoneNumber, preferredLanguage };
+      const patch = (await request.json()) as Operation[];
+      const patchResult = jsonpatch.applyPatch(document, patch, true);
+      db.user.update({ where: { id: { equals: id } }, data: patchResult.newDocument });
+    } catch (error) {
+      if (error instanceof JsonPatchError) {
+        return HttpResponse.text(error.message, { status: 400 });
```

Comment:
![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/089dc20e-7d15-4ef4-b175-7ae2383a4438)

Reply:
No issue on my computer.

Reply:
My problem.. forgot to `npm install`. Resolving.